src/storage/StorageConfig.js

```diff
@@ -0,0 +1,91 @@
+const fetch = require('node-fetch')
+
+const logger = require('../helpers/logger')('streamr:storage:StorageConfig')
+
+/*
+ * Connects to Core API and queries the configuration there.
+ * Refreshes the config at regular intervals.
+ */
+const getStreamFromKey = (key) => {
+    const [id, partitionStr] = key.split('::')
+    return {
+        id,
+        partition: Number(partitionStr)
+    }
+}
+
+const getKeyFromStream = (streamId, streamPartition) => {
+    return `${streamId}::${streamPartition}`
+}
+
+module.exports = class StorageConfig {
+    // use createInstance method instead: it fetches the up-to-date config from API
+    constructor(nodeId, apiUrl) {
+        this.streams = new Set()
+        this.listeners = []
```

Comment:
You could, I guess, get rid of maintaining this `listeners` array by extending `EventEmitter` and emitting events instead?

Reply:
We can refactor this later